rfc/rfc-0009-controller-unit-testing.md

```diff
@@ -0,0 +1,84 @@
+# RFC-0009: Controller Unit Testing
+
+**Authors:** Glyn Normington
+
+**Status:**
+
+**Pull Request URL:** https://github.com/projectriff/riff/pull/1372
+
+**Superseded by:** N/A
+
+**Supersedes:** N/A
+
+**Related:** N/A
+
+
+## Problem
+The Kubernetes controllers in riff need unit testing in a consistent way. Other projects
+have adopted and promoted various ways of testing controllers. For ease of maintenance,
+riff needs to adopt a single, consistent approach.
+
+For the purpose of this document, unit testing refers to testing the
+`Reconcile` method of a controller.
+
+Apart from checking that various objects are created etc., it should be straigthtforward
+to test error paths by injecting errors in object lookup and creation.
+
+### Anti-Goals
+
+None.
```

Comment:
what about the execution mode (as in serial vs. parallel)? It feels out of scope for this spec (hence an anti-goal) but could be specified later?

Reply:
is it out of scope to talk about the developer experience of the test runner here? 
I'm thinking about the ability to focus on/skip a test, error reporting when tests fail.

Reply:
or, wrapping the 2 previous comments, that an anti-goal of this RFC is to specify the capabilities of the underlying custom test runner

Reply:
Yes, I'll add an anti-goal.